packages/filesystem/src/browser/file-resource.ts

```diff
@@ -65,8 +65,8 @@ export class FileResource implements Resource {
     get encoding(): string | undefined {
         return this._version?.encoding;
     }
-    get readOnly(): boolean {
-        return this.options.isReadonly || this.fileService.hasCapability(this.uri, FileSystemProviderCapabilities.Readonly);
+    get readOnly(): boolean | MarkdownString {
+        return this.options.readOnly;
```

Comment:
This introduces a regression: Opening a file in a readonly file system marks the file incorrectly as writable, since the file options are decoupled from the capabilities of the file system when creating the initial `FileResource`. See also:

https://github.com/eclipse-theia/theia/blob/31632178a156904eb478259c8bb4dabee597b54d/packages/filesystem/src/browser/file-resource.ts#L350-L354

Reply:
I updated the way readOnly was setup. This regression should now be removed.